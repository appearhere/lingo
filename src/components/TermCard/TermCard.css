.root {
  padding: calc(var(--size-lg-i) * 2);
  position: relative;
  background-color: var(--color-white);
}

.root:before {
  content: "";
  display: block;
  padding-top: 100%;
  float: left;
}

.root > a {
  text-decoration: none;
}

.name {
  font-size: var(--fontsize-large-ii);
  font-weight: var(--fontweight-demi);
  color: var(--color-black);
}

.department {
  position: absolute;
  bottom: calc(var(--size-lg-i) * 2);
  font-size: var(--fontsize-small-i);
  font-weight: var(--fontweight-regular);
  color: var(--color-black);
  text-transform: uppercase;
}
