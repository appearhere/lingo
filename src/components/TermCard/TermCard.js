import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import cardCss from '@appearhere/bloom/components/Cards/Card/Card.css';
import css from './TermCard.css';

const TermCard = ({ term, className }) => (
  <div className={ cx(css.root, cardCss.link, className) }>
    <Link to={ `/${term.department}/${term.name}` }>
      <div className={ css.name }>{ term.name }</div>
      <div className={ css.department }>{ term.department }</div>
    </Link>
  </div>
);

TermCard.propTypes = {
  term: PropTypes.shape({}).isRequired,
  className: PropTypes.string,
};

export default TermCard;
