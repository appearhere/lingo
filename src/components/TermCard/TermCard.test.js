import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';

import TermCard from './TermCard';

describe('TermCard component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StaticRouter><TermCard term={ {} } /></StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
