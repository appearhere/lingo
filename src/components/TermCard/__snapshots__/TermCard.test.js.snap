// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermCard component renders correctly 1`] = `
<a
  className=""
  href="/undefined/undefined"
  onClick={[Function]}
>
  <div
    className={undefined}
  />
  <div
    className={undefined}
  />
</a>
`;
